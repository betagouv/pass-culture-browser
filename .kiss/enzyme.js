import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
