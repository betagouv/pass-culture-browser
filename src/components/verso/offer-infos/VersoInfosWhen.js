import React from 'react'
import get from 'lodash.get'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

class VersoInfosWhen extends React.PureComponent {
  renderEventOfferDateInfos() {
    const { bookables, maxDatesShowned } = this.props
    const sliced = bookables && bookables.slice(0, maxDatesShowned)
    const hasMoreBookables = bookables && bookables.length > maxDatesShowned
    return (
      <React.Fragment>
        {sliced &&
          sliced.map(obj => (
            <li key={obj.id}>
              {capitalize(obj.humanBeginningDate)}
              {obj.userAsAlreadyReservedThisDate && ' (réservé)'}
            </li>
          ))}
        {hasMoreBookables && (
          <li>{'Cliquez sur "j\'y vais" pour voir plus de dates.'}</li>
        )}
      </React.Fragment>
    )
  }

  renderThingOfferDateInfos = () => {
    const { bookables } = this.props
    const limitDatetime =
      bookables && get(bookables, '[0].bookinglimitDatetime')
    return (
      <React.Fragment>
        <li>
          Dès maintenant 
          {' '}
          {limitDatetime && `et jusqu&apos;au ${limitDatetime}`}
          {' '}
        </li>
      </React.Fragment>
    )
  }

  render() {
    const { bookables, isFinished, recommendation } = this.props
    const dateInfosRenderer = (get(recommendation, 'offer.thingId')
      ? this.renderThingOfferDateInfos
      : this.renderEventOfferDateInfos
    ).bind(this)
    return (
      <div>
        <h3>Quand ?</h3>
        <ul className="dates-info">
          {isFinished ? (
            <li>L&apos;offre n&apos;est plus disponible :(</li>
          ) : (
            dateInfosRenderer()
          )}
        </ul>
      </div>
    )
  }
}

VersoInfosWhen.defaultProps = {}

VersoInfosWhen.propTypes = {
  bookables: PropTypes.array,
  isFinished: PropTypes.bool,
  recommendation: PropTypes.object,
}

export default VersoInfosWhen
