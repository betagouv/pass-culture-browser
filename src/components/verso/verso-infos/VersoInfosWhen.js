import React from 'react'
import get from 'lodash.get'
import PropTypes from 'prop-types'
import { capitalize } from 'pass-culture-shared'
// import { connect } from 'react-redux'

class VersoInfosWhen extends React.PureComponent {
  renderEventOfferDateInfos = () => {
    const { bookables, maxDatesShowned } = this.props
    const sliced = bookables && bookables.slice(0, maxDatesShowned)
    const hasMoreBookables = bookables && bookables.length > maxDatesShowned
    return (
      <React.Fragment>
        {sliced &&
          sliced.map(obj => (
            <li key={obj.id}>
              {capitalize(obj.humanBeginningDate)}
              {obj.userAsAlreadyReservedThisDate && ' (réservé)'}
            </li>
          ))}
        {hasMoreBookables && (
          <li>{'Cliquez sur "j\'y vais" pour voir plus de dates.'}</li>
        )}
      </React.Fragment>
    )
  }

  renderThingOfferDateInfos = () => {
    const { bookables } = this.props
    const limitDatetime =
      bookables && get(bookables, '[0].bookinglimitDatetime')
    return (
      <React.Fragment>
        <li>
          Dès maintenant 
          {' '}
          {limitDatetime && `et jusqu&apos;au ${limitDatetime}`}
          {' '}
        </li>
      </React.Fragment>
    )
  }

  render() {
    const { isFinished } = this.props
    const { recommendation } = this.props
    const dateInfosRenderer = get(recommendation, 'offer.thingId')
      ? this.renderThingOfferDateInfos
      : this.renderEventOfferDateInfos
    return (
      <div id="verso-infos-section-when">
        <h3>Quand ?</h3>
        <ul className="dates-info">
          {isFinished ? (
            <li>L&apos;offre n&apos;est plus disponible :(</li>
          ) : (
            dateInfosRenderer()
          )}
        </ul>
      </div>
    )
  }
}

VersoInfosWhen.defaultProps = {
  bookables: null,
  isFinished: false,
  maxDatesShowned: 7,
  recommendation: null,
}

VersoInfosWhen.propTypes = {
  bookables: PropTypes.array,
  isFinished: PropTypes.bool,
  maxDatesShowned: PropTypes.number,
  recommendation: PropTypes.object,
}

export default VersoInfosWhen
