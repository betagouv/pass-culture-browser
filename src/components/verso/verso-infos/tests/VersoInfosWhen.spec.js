// jest --env=jsdom ./src/components/verso/tests/VersoInfosWhen.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'
import VersoInfosWhen from '../VersoInfosWhen'

describe('src | components | VersoInfosWhen', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {}
      // when
      const wrapper = shallow(<VersoInfosWhen {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
