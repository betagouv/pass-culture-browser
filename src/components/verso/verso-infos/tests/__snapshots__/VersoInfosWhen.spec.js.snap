// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | VersoInfosWhen snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VersoInfosWhen
    bookables={null}
    isFinished={false}
    maxDatesShowned={7}
    recommendation={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Quand ?
        </h3>,
        <ul
          className="dates-info"
        >
          <React.Fragment />
        </ul>,
      ],
      "id": "verso-infos-section-when",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Quand ?",
        },
        "ref": null,
        "rendered": "Quand ?",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <React.Fragment />,
          "className": "dates-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              null,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
          ],
          "type": Symbol(react.fragment),
        },
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Quand ?
          </h3>,
          <ul
            className="dates-info"
          >
            <React.Fragment />
          </ul>,
        ],
        "id": "verso-infos-section-when",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Quand ?",
          },
          "ref": null,
          "rendered": "Quand ?",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <React.Fragment />,
            "className": "dates-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                null,
                null,
              ],
            },
            "ref": null,
            "rendered": Array [
              null,
              null,
            ],
            "type": Symbol(react.fragment),
          },
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
